metaserver/src/main/scala/scala/meta/languageserver/ctags/ScalaCtags.scala

```diff
@@ -0,0 +1,35 @@
+package scala.meta.languageserver.ctags
+
+import scala.meta._
+import org.langmeta.inputs.Input
+
+object ScalaCtags {
+  def index(input: Input.VirtualFile): CtagsIndexer = {
+    val root: Source = input.parse[Source].get
+    new Traverser with CtagsIndexer {
+      override def language: String = "Scala212" // TODO(olafur) more accurate dialect
+      override def indexRoot(): Unit = apply(root)
+      override def apply(tree: Tree): Unit = {
+        val old = currentOwner
+        val next = tree match {
+          case t: Source => Continue
+          case t: Template => Continue
+          case t: Pkg => pkg(t.ref); Continue
+          case t: Pkg.Object => term(t.name, PACKAGEOBJECT); Continue
+          case t: Defn.Class => tpe(t.name, CLASS); Continue
+          case t: Defn.Trait => tpe(t.name, TRAIT); Continue
+          case t: Defn.Object => term(t.name, OBJECT); Continue
+          case t: Defn.Def => term(t.name, DEF); Stop
+          case Defn.Val(_, Pat.Var(name) :: Nil, _, _) => term(name, DEF); Stop
+          case Defn.Var(_, Pat.Var(name) :: Nil, _, _) => term(name, DEF); Stop
```

Comment:
@olafurpg What about declarations? For example if we have somewhere defined

```
trait T { 
  type X
  val x: X 
}
```

`X` will be a `Decl.Type` and `x` a `Decl.Val`, so if later in the code we encounter something like `def foo(t: T): t.X` and want to lookup that `t.X`... No, this example is bad, but I think you got the idea. If not, I'll think of a better example, sorry 😅

Reply:
Those are actually not handled! This is a bug 😅

Reply:
OK, I'll try to add it then. Another question here

Reply:
> Do you think it could be also useful for the general indexer to use that wider position range? e.g.

Yes! I've been considering refactoring the indexer to construct a custom data structure that we can tweak for our needs. Then we can reuse it for multiple features like outline + go to definition.

Can you open a ticket to discuss details?